dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassDumperTest.java

```diff
@@ -364,4 +366,26 @@ public void testFindSymbolReferences_overLappingClass() throws IOException, Repo
             new ClassFile(
                 sisuGuicePath, "com.google.inject.internal.InjectorImpl$BindingsMultimap"));
   }
+
+  @Test
+  public void testListClasses_unexpectedNonClassFile() throws RepositoryException, IOException {
+    // com.amazonaws:amazon-kinesis-client:1.13.0 contains an unexpected lock file
+    // /unison/com/e007f77498fd27177e2ea931a06dcf50/unison/tmp/amazonaws/services/kinesis/leases/impl/LeaseTaker.class
+    // https://github.com/awslabs/amazon-kinesis-client/issues/654
+    Artifact kinesisClient = new DefaultArtifact("com.amazonaws:amazon-kinesis-client:1.13.0");
+    List<Path> paths = ClassPathBuilder.artifactsToClasspath(ImmutableList.of(kinesisClient));
+    ClassDumper classDumper = ClassDumper.create(paths);
+    Path kinesisJar = paths.get(0);
+
+    // This should not raise IOException
+    SymbolReferenceMaps symbolReferences = classDumper.findSymbolReferences();
+    assertNotNull(symbolReferences);
+
+    Truth.assertWithMessage("Invalid files should not stop loading valid class files")
+        .that(symbolReferences.getClassToClassSymbols().keySet())
+        .comparingElementsUsing(
+            Correspondence.transforming(
+                (ClassFile classFile) -> classFile.getJar(), "is in the JAR file"))
+        .contains(kinesisJar);
```

Comment:
This asserts that ClassDumper loads valid class files from the JAR file even if some of the content are invalid files.